GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/WorkingDirectoryTests.cs

```diff
@@ -303,45 +303,36 @@ public void EnumerateAndReadDoesNotChangeEnumerationOrder()
             folderVirtualPath.ShouldBeADirectory(this.fileSystem).WithItems();
         }
 
+        // Case-insensitive filesystems only
         [TestCase, Order(7)]
-        public void HydratingFileUsesNameCaseFromRepo()
+        [Category(Categories.CaseInsensitiveFileSystemOnly)]
+        public void HydratingFileUsesNameCaseFromRepoOnCaseInsensitiveFileSystem()
         {
-            string fileName = "Readme.md";
-            string parentFolderPath = this.Enlistment.GetVirtualPathTo(Path.GetDirectoryName(fileName));
-            parentFolderPath.ShouldBeADirectory(this.fileSystem).WithItems().ShouldContainSingle(info => info.Name.Equals(fileName, StringComparison.Ordinal));
-
-            // Hydrate file with a request using different file name case
-            string wrongCaseFilePath = this.Enlistment.GetVirtualPathTo(fileName.ToUpper());
-            string fileContents = wrongCaseFilePath.ShouldBeAFile(this.fileSystem).WithContents();
+            this.HydratingFileUsesNameCaseFromRepo(false);
+        }
 
-            // File on disk should have original case projected from repo
-            parentFolderPath.ShouldBeADirectory(this.fileSystem).WithItems().ShouldContainSingle(info => info.Name.Equals(fileName, StringComparison.Ordinal));
+        // Case-sensitive filesystems only
+        [TestCase, Order(7)]
+        [Category(Categories.CaseSensitiveFileSystemOnly)]
+        public void HydratingFileUsesNameCaseFromRepoOnCaseSensitiveFileSystem()

```

Comment:
Rather than having two separate methods here tagged as `CaseInsensitiveFileSystemOnly` and `CaseSensitiveFileSystemOnly` could we go back to having a single method that would check if the current filesystem is case-sensitive?

In other words, there could be a single `HydratingFileUsesNameCaseFromRepo` test with a line like this

`string testFileName = FileSystemHelpers.CaseSensitiveFileSystem ? fileName : fileName.ToUpper();`

Reply:
Much better!  Thank you!